src/plugins/fs/generic.c

```diff
@@ -609,6 +650,104 @@ gboolean bd_fs_set_label (const gchar *device, const gchar *label, GError **erro
     return device_operation (device, BD_FS_LABEL, 0, label, error);
 }
 
+static BDFSXfsInfo* xfs_get_info (const gchar *device, GError **error) {
+    g_autofree gchar* mountpoint = NULL;
+    gboolean unmount = FALSE;
+    gboolean ret = FALSE;
+    GError *local_error = NULL;
+    BDFSXfsInfo* xfs_info = NULL;
+
+    mountpoint = xfs_mount (device, &unmount, error);
+    if (!mountpoint)
+        return NULL;
+
+    xfs_info = bd_fs_xfs_get_info (device, error);
+
+    if (unmount) {
+        ret = bd_fs_unmount (mountpoint, FALSE, FALSE, NULL, &local_error);
+        if (!ret) {
+            bd_utils_log_format (BD_UTILS_LOG_INFO, "Failed to unmount %s after getting information about it: %s", device, local_error->message);
+            g_clear_error (&local_error);
+        } else
+            g_rmdir (mountpoint);
+    }
+
+    return xfs_info;
+}
+
+/**
+ * bd_fs_get_size:
+ * @device: the device with file system to get size for
+ * @error: (out): place to store error (if any)
+ *
+ * Get size for filesystem on @device. This calls other fs info functions from this
+ * plugin based on detected filesystem (e.g. bd_fs_xfs_get_info for XFS). This
+ * function will return an error for unknown/unsupported filesystems.
+ *
+ * Returns: size of filesystem on @device, 0 in case of error.
```

Comment:
I'm not sure if it's safe to use `0` as a "error" value. Can we have a filesystem with zero size? Alternatively we could return boolean and return the size in an output variable.

Reply:
`0` is OK, I think. I don't think there can be an FS with zero size.